### PR TITLE
feat: Add TypeScript language analyzer

### PR DESCRIPTION
**Pull request by WWykret**

### CODE REVIEW

File: backend/fregepoc/analyzers/core/typescript.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+class TypescriptFileAnalysisResult(TypedDict):
+    lines_of_code: int
+    token_count: int
+    average_lines_of_code: float
+    average_token_count: float
+    average_cyclomatic_complexity: float
+
+
+@AnalyzerFactory.register(ProgrammingLanguages.TYPESCRIPT)
+class JavascriptAnalyzer(BaseAnalyzer[TypescriptFileAnalysisResult]):
```
Comment (vladfedoriuk) [inline backend/fregepoc/analyzers/core/typescript.py:19]: I guess `TypescriptAnalyzer` could be a better name for the class :)

File: backend/fregepoc/analyzers/tests/typescript/test_typescript_avarage_cyclomatic_complexity.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                {"repo_relative_file_path": "bubble_sort.ts"},
+                2.0,
+            ),
+            (
+                {"repo_relative_file_path": "fast_fibbonaci.ts"},
+                2.5,
+            ),
+        ],
+    )
+    def test_count_loc(
```
Comment (vladfedoriuk) [inline backend/fregepoc/analyzers/tests/typescript/test_typescript_avarage_cyclomatic_complexity.py:34]: Is there a reason for not including all those tests in a single 'Test Case' class? I think all those test classes could have been merged into one.

File: backend/fregepoc/analyzers/core/typescript.py (hunk 1 above)
Comment (vladfedoriuk) [reply backend/fregepoc/analyzers/core/typescript.py:19]: Also, how is this different from JavaScript analyzer? Can't we use the same analyzer for both languages? If there are no differences, I don't see a point of having that class duplicated. Are there any TS-specific metrics? Could you add some for this class?

File: backend/fregepoc/analyzers/tests/typescript/constants.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from fregepoc import settings
+
+MOCKED_TYPESCRIPT_FILES = (
```
Comment (vladfedoriuk) [inline backend/fregepoc/analyzers/tests/typescript/constants.py:3]: Why have you decided to change a default location of testing files? Most of the analyzers have their testing files in the `repositories` app. Also,  there has already  been defined a mock path for that location. I would recommend sticking to that convention, unless you have some reasons for not doing it.

File: backend/fregepoc/repositories/constants.py [hunk 4]
```diff
@@ -67,3 +68,4 @@ def get_languages_by_extension(
 register_extension("KOTLIN", [".kt"])
 register_extension("RUST", [".rs"])
 register_extension("GOLANG", [".go"])
+register_extension("TYPESCRIPT", [".ts", ".tsx"])
```
Comment (vladfedoriuk) [inline backend/fregepoc/repositories/constants.py:71]: I would rather be careful about adding `tsx` extension here. Are you sure the metrics used in analyzer would cope with understanding JSX?

File: backend/fregepoc/analyzers/core/typescript.py (hunk 1 above)
Comment (WWykret) [reply backend/fregepoc/analyzers/core/typescript.py:19]: I can't really think of any TS-specific metrics that wouldn't apply to JS, however lizard (which both TS and JS analyzers use) treats this languages differently, hence I belive it is problematic to use one analyzer for both JS and TS

File: backend/fregepoc/analyzers/tests/typescript/test_typescript_avarage_cyclomatic_complexity.py (hunk 2 above)
Comment (WWykret) [reply backend/fregepoc/analyzers/tests/typescript/test_typescript_avarage_cyclomatic_complexity.py:34]: They probably could be easily merged, however I am following convention present in every other analyzers tests

File: backend/fregepoc/analyzers/tests/typescript/constants.py (hunk 3 above)
Comment (WWykret) [reply backend/fregepoc/analyzers/tests/typescript/constants.py:3]: I believe the files you mean aren't meant for actual unit testing of analyzers, but rather downloading repos. I was following convention used in other analyzers where they keep files for unit tests in analogous location

File: backend/fregepoc/repositories/constants.py (hunk 4 above)
Comment (WWykret) [reply backend/fregepoc/repositories/constants.py]: You are right, as it turns out lizard doesn't support .tsx files yet.

File: backend/fregepoc/analyzers/core/__init__.py [hunk 5]
```diff
@@ -1,8 +1,9 @@
 from .cpp import *  # noqa
+from .golang import *  # noqa
```
Comment (vladfedoriuk) [inline backend/fregepoc/analyzers/core/__init__.py:2]: Actually, one more minor thing - please, make those imports absolute and add a specific error code to ignore in `# noqa` for all the imports.

Comment (WWykret) [reply backend/fregepoc/analyzers/core/__init__.py]: I made imports absolute, and removed `#noqa` altogether.

Comment (JaroslawHryszko) [reply backend/fregepoc/analyzers/core/__init__.py]: @vladfedoriuk if you think that no additional changes are needed, please merge these changes :)

**Pull request merged**